dbms/src/Common/Stopwatch.h

```diff
@@ -23,21 +24,28 @@
 #include <common/apple_rt.h>
 #endif
 
-
-namespace StopWatchDetail
-{
-inline UInt64 nanoseconds(clockid_t clock_type)
+inline UInt64 clock_gettime_ns(clockid_t clock_type = CLOCK_MONOTONIC)
 {
-    struct timespec ts;
+    struct timespec ts
+    {
+    };
     clock_gettime(clock_type, &ts);
-    return ts.tv_sec * 1000000000ULL + ts.tv_nsec;
+    return UInt64(ts.tv_sec * 1000000000LL + ts.tv_nsec);
 }
-inline UInt64 seconds(clockid_t clock_type)
+
+/// Sometimes monotonic clock may not be monotonic (due to bug in kernel?).
+/// It may cause some operations to fail with "Timeout exceeded: elapsed 18446744073.709553 seconds".
+/// Takes previously returned value and returns it again if time stepped back for some reason.
+inline UInt64 clock_gettime_ns_adjusted(UInt64 prev_time, clockid_t clock_type = CLOCK_MONOTONIC)
 {
-    return nanoseconds(clock_type) / 1000000000ULL;
-}
-} // namespace StopWatchDetail
+    UInt64 current_time = clock_gettime_ns(clock_type);
+    if (likely(prev_time <= current_time))
+        return current_time;
 
+    /// Something probably went completely wrong if time stepped back for more than 1 second.
+    assert(prev_time - current_time <= 1000000000ULL);
+    return prev_time;
```

Comment:
`return std::max(prev_time, current_time)`?

Reply:
good idea!

Reply:
sorry, I found that the previous version
```
if (likely(prev_time <= current_time))
    return current_time;

/// Something probably went completely wrong if time stepped back for more than 1 second.
assert(prev_time - current_time <= 1000000000ULL);
return prev_time;
```
seems to be more effective than `std::max`.